src/components/ProtectedRoutes/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { Redirect, Route, Switch } from 'react-router-dom';
+
+import Login from 'screens/Login';
+import Signup from 'screens/Signup';
+import Home from 'screens/Home';
+
+function ProtectedRoutes() {
+  const isLoggedIn = false;
+
+  return (
+    <Switch>
+      <Route path="/sign_up">{isLoggedIn ? <Redirect to="/" /> : <Signup />}</Route>
+      <Route path="/">{isLoggedIn ? <Home /> : <Login />}</Route>
```

Comment:
This structure may turn cumbersome when more routes are added.
Imagine your routes as a list of: which URL is shown, which screen is shown along with that url, should the user access that router always or based on a condition, if not should I redirect him somewhere else?